lib/rubygems.rb

```diff
@@ -800,7 +800,7 @@ def self.read_binary(path)
   ##
   # Safely write a file in binary mode on all platforms.
   def self.write_binary(path, data)
-    File.open(path, File::RDWR | File::CREAT | File::BINARY | File::LOCK_EX) do |io|
+    File.open(path, File::RDWR | File::CREAT | File::LOCK_EX, binmode: true) do |io|
```

Comment:
I think this change only fixed the race condition because it made the file write more atomic, but the file lock was not having any effect anyways, because the reading side (`Gem::Specification.load`) does not use the lock. I guess we have actually unintentionally removed any file locking from this method, so if nobody else complains I guess we can completely cleanup it up and remove all traces of file locking in the future. 🤷‍♂️